### PR TITLE
Python Docker Tutorial - Bodge to make it functional (Bash & PowerShell)

### PR DESCRIPTION
**Pull request by Gofven**
Thanks everyone for making this tutorial, but multiple parts of the tutorial doesn't work due to inconsistency between each step, so I bodged up a solution to make it work. I am still working on improving it, help would be much appreciated!

(Keep in mind MYSQL_ALLOW_EMPTY_PASSWORD was passed in shell at line 41, so I adjusted the page to follow that. (Passing passwords trough environments isnt safe eitherway))

### Proposed changes
    + Adding functionality to Python Tutorial, the master branch at it's current state is broken no matter who uses it.
    + Support for PowerShell users

### CODE REVIEW

**Comment by usha-mandya** [comment]
@StefanScherer Could you PTAL?

**Comment by Gofven** [comment]
I'm still working on it. There's still plenty left to edit, and I'll try to comment the changes as clear as I possibly could. The push should be ready in a week or so

**Comment by usha-mandya** [comment]
@Gofven This is great. We really appreciate your help in improving the docs.

File: language/python/build-images.md [hunk 1]
```diff
@@ -25,10 +25,19 @@ To complete this tutorial, you need the following:
...
 
 ```shell
-$ cd /path/to/python-docker
-$ pip3 install Flask
-$ pip3 freeze > requirements.txt
-$ touch app.py
+cd /path/to/python-docker
+pip3 install Flask
+pip3 freeze > requirements.txt
+touch app.py
```
Comment (StefanScherer) [inline language/python/build-images.md:31]: I suggest to remove this `touch app.py` line. Then we won't need the PowerShell section.

Comment (StefanScherer) [reply language/python/build-images.md:31]: In all other places in our docs we show commands normally with a shell prompt `$`. I know in PowerShell it looks different, but we prefer to keep it to be consistent to the rest of the docs.

File: language/python/build-images.md [hunk 2]
```diff
@@ -47,7 +56,7 @@ def hello_world():
 Let’s start our application and make sure it’s running properly. Open your terminal and navigate to the working directory you created.
 
 ```shell
-$ python3 -m flask run
+python3 -m flask run
```
Comment (StefanScherer) [inline language/python/build-images.md:59]: Please keep the prompt and in following shell blocks.

File: language/python/build-images.md [hunk 3]
```diff
@@ -96,10 +105,10 @@ Before we can run `pip3 install`, we need to get our `requirements.txt` file int
 COPY requirements.txt requirements.txt
 ```
 
-Once we have our `requirements.txt` file inside the image, we can use the `RUN` command to execute the command `pip3 install`. This works exactly the same as if we were running `pip3 install` locally on our machine, but this time the modules are installed into the image.
+Once we have our `requirements.txt` file inside the image, we can use the `RUN` command to execute the command `python3 -m pip install`. This works exactly the same as if we were running `python3 -m pip install` locally on our machine, but this time the modules are installed into the image.
 
 ```dockerfile
-RUN pip3 install -r requirements.txt
+RUN python3 -m pip install -r requirements.txt
```
Comment (StefanScherer) [inline language/python/build-images.md:111]: I'm not a day to day Python developer, but this looks a bit weird to me as inside a container `pip3` is installed.
I understand that you want to have it consistent to the commands to run on the host, but do Python developers actually run pip this way? Just asking as I'm curious 😃

File: language/python/build-images.md [hunk 4]
```diff
@@ -167,7 +187,7 @@ $ docker build --tag python-docker .
  => => exporting layers
  => => writing image sha256:8cae92a8fbd6d091ce687b71b31252056944b09760438905b726625831564c4c
  => => naming to docker.io/library/python-docker
-```
+ ```
```
Comment (StefanScherer) [inline language/python/build-images.md:190]: There is a space that should not be there.

File: language/python/build-images.md [hunk 5]
```diff
@@ -149,7 +164,12 @@ The build command optionally takes a `--tag` flag. The tag is used to set the na
 Let’s build our first Docker image.
 
 ```shell
-$ docker build --tag python-docker .
+docker build --tag python-docker .
```
Comment (StefanScherer) [inline language/python/build-images.md:167]: We prefer one code block with a shell prompt and the output together.

File: language/python/develop.md [hunk 6]
```diff
@@ -22,7 +22,7 @@ Instead of downloading MySQL, installing, configuring, and then running the MySQ
 
 Before we run MySQL in a container, we'll create a couple of volumes that Docker can manage to store our persistent data and configuration. Let’s use the managed volumes feature that Docker provides instead of using bind mounts. You can read all about [Using volumes](../../storage/volumes.md) in our documentation.
 
-Let’s create our volumes now. We’ll create one for the data and one for configuration of MongoDB.
+Let’s create our volumes now. We’ll create one for the data and one for configuration of MySQL.
```
Comment (StefanScherer) [inline language/python/develop.md:25]: Thanks!

File: language/python/develop.md [hunk 7]
```diff
@@ -147,7 +165,7 @@ First, let’s add the `mysql-connector-python `module to our application using
 
 ```shell
 $ pip3 install mysql-connector-python
-$ pip3 freeze -r requirements.txt
+$ pip3 freeze > requirements.txt
```
Comment (StefanScherer) [inline language/python/develop.md:168]: Thanks!

File: language/python/run-containers.md [hunk 8]
```diff
@@ -19,20 +19,20 @@ A container is a normal operating system process except that this process is iso
...
 Let’s make a `GET` request to the server using the `curl` command.
 
 ```shell
-$ curl --request POST \
-  --url http://localhost:8000/test \
-  --header 'content-type: application/json' \
-  --data '{
-	"msg": "testing"
-}'
+curl http://localhost:8000/
```
Comment (StefanScherer) [inline language/python/run-containers.md:30]: When looking at the tutorial I thought we should just go with port 5000 everywhere.

File: language/python/run-containers.md [hunk 9]
```diff
@@ -45,19 +45,19 @@ To publish a port for our container, we’ll use the `--publish flag` (`-p` for
 Start the container and expose port 8000 to port 8000 on the host:
 
 ```shell
-$ docker run --publish 8000:8000 python-docker
+docker run --publish 8000:8000 python-docker
```
Comment (StefanScherer) [inline language/python/run-containers.md:48]:
```suggestion
$ docker run --publish 5000:5000 python-docker
```

File: language/python/run-containers.md [hunk 10]
```diff
@@ -73,7 +73,12 @@ Press ctrl-c to stop the container.
 This is great so far, but our sample application is a web server and we don't have to be connected to the container. Docker can run your container in detached mode or in the background. To do this, we can use the `--detach` or `-d` for short. Docker starts your container the same as before but this time will “detach” from the container and return you to the terminal prompt.
 
 ```shell
-$ docker run -d -p 8000:8000 python-docker
+docker run -d -p 8000:8000 python-docker
```
Comment (StefanScherer) [inline language/python/run-containers.md:76]:
```suggestion
$ docker run -d -p 5000:5000 python-docker
```

File: language/python/run-containers.md [hunk 11]
```diff
@@ -82,21 +87,26 @@ Docker started our container in the background and printed the Container ID on t
...
 
 ```shell
-$ curl --request POST \
-  --url http://localhost:8000/test \
-  --header 'content-type: application/json' \
-  --data '{
-	"msg": "testing"
-}'
-{"code":"success","payload":[{"msg":"testing","id":"dc0e2c2b-793d-433c-8645-b3a553ea26de","createDate":"2020-09-01T17:36:09.897Z"}]}
+curl http://localhost:8000/
```
Comment (StefanScherer) [inline language/python/run-containers.md:90]:
```suggestion
$ curl http://localhost:5000/
```

**Review by StefanScherer — COMMENTED** [review]
Thanks @Gofven for your contribution and fixing the Python tutorial.

I left a couple of review comments. It would be great if you could have a look at them and adjust the PR a bit to our needs. If it's too much let me know, then we could carry this PR and tidy up some bits.

**Comment by StefanScherer** [comment]
Hello @Gofven, we updated our Python tutorial to fix it. It will go live soon, the changes are in #12559. Let me know how we could improve it for PowerShell users.

File: language/python/build-images.md (hunk 1 above)
Comment (Sammeeey) [reply language/python/build-images.md:31]: The `touch` command doesn't work for windows anyway - as I told here: #13124 
Maybe it's better to make that clear and to tell people to create a file in a different way on windows?

**Comment by usha-mandya** [comment]
@Gofven It's been more than a year since the PR is in the Draft state. Closing this PR for this reason. Please feel free to reopen with the updated changes if it's still relevant. Thank you.

**Pull request closed without merging**